src/features/ProfilePage/components/PublicInfo/index.tsx

```diff
@@ -69,6 +72,29 @@ const PublicInfo = ({ isMobile = false }: Props) => {
     }
   };
 
+  // Block navigating elsewhere in the app if there are unsaved changes
+  const blocker = useBlocker(() => isDirty);
+
+  useEffect(() => {
+    const confirmLeave = async () => {
+      const isConfirmed = await getConfirmation({
+        borderColor: palette.redSalmon,
+        closeText: t('public.mentor.leavePage.cancel'),
+        confirmId: 'confirm-leave',
+        confirmText: t('public.mentor.leavePage.confirm'),
+        description: t('public.mentor.leavePage.description'),
+        title: t('public.mentor.leavePage.title'),
+      });
+      if (isConfirmed) {
+        if (blocker.state === 'blocked') blocker.proceed();
+      } else {
+        if (blocker.state === 'blocked') blocker.reset();
+      }
+    };
+
+    if (blocker.state === 'blocked') confirmLeave();
+  }, [blocker.state]);
```

Comment:
This could be abstracted to Confirmation-feature, as its own hook like "useConfirmDirtyLeave" etc

Reply:
Good suggestion! 💪 I'll implement this